rp2040-hal/src/timer.rs

```diff
@@ -340,6 +335,18 @@ macro_rules! impl_alarm {
                 (bits & $armed_bit_mask) == 0
             }
         }
+
+        impl Drop for $name {
+            fn drop(&mut self) {
+                self.disable_interrupt();
+                release_alarm($armed_bit_mask)
+            }
+        }
+
+        // safety: there can only be only 1 instance of this alarm so no shared mutable access
+        unsafe impl Send for $name {}
+        // safety: all write operations are synchronised and all reads are atomic
+        unsafe impl Sync for $name {}
```

Comment:
AFAIK Alarm* already implement Send and Sync even without explicit declaration, as those are auto traits.
https://docs.rs/rp2040-hal/latest/rp2040_hal/timer/struct.Alarm0.html#impl-Send-for-Alarm0